runtime/bcutil/ComparingCursor.cpp

```diff
@@ -212,10 +212,13 @@ ComparingCursor::writeSRP(UDATA srpKey, DataType dataType)
 					markUnEqual();
 				} else {
 					U_16 cpIndex = _srpKeyProducer->mapKeyToCfrConstantPoolIndex(srpKey);
+					if ( cpIndex > _classFileOracle->getConstantPoolCount() ) {
```

Comment:
needs ifdef

Reply:
should this be `>=` ?

Reply:
also add a comment indicating that cp index's aboive the cpCount are injected interfaces which should not be written into the classfile bytes